spec/lib/registration-manager.spec.js

```diff
@@ -202,85 +213,94 @@ describe("registrationManager", () => {
         beforeEach(() => {
             serverMock = require("../mock/server-mock")();
         });
-        it("fails on invalid TOTP keys", (done) => {
+        it("fails on invalid TOTP keys", () => {
             totpMock.verifyCurrentAndPrevious.andReturn(false);
-            factory().secureAsync("id", {
-                totp: {
-                    current: "000000",
-                    previous: "111111"
+
+            return factory().secureAsync("id", {
+                mfa: {
+                    totp: {
+                        current: "000000",
+                        previous: "111111"
+                    }
                 }
-            }, serverMock).then(() => {
-                jasmine.fail();
-                done();
-            }, (err) => {
+            }, serverMock).then(jasmine.fail, (err) => {
                 expect(totpMock.verifyCurrentAndPrevious).toHaveBeenCalledWith("totp key", "000000", "111111");
                 expect(storageService.putAsync).not.toHaveBeenCalled();
                 expect(err).toEqual(jasmine.any(Error));
                 expect(err.toString()).toContain("TOTP validation failed");
-                done();
             });
         });
-        it("works", (done) => {
-            factory().secureAsync("id", {
-                passwordHash: "hashhash",
-                totp: {
-                    current: "000000",
-                    previous: "111111"
-                }
+        it("works", () => {
```

Comment:
We should be a little more descriptive: "succeeds with valid TOTP values", something like that.